go-controller/pkg/ovn/namespace.go

```diff
@@ -199,7 +201,23 @@ func (oc *Controller) AddNamespace(ns *kapi.Namespace) {
 			nsInfo.hybridOverlayVTEP = parsedAnnotation
 		}
 	}
-
+	annotation = ns.Annotations[routingExternalGWsAnnotation]
+	if annotation != "" {
+		annotationOk := true
+		for _, v := range strings.Split(annotation, ",") {
+			parsedAnnotation := net.ParseIP(v)
+			if parsedAnnotation == nil {
+				klog.Errorf("Could not parse routing external gw annotation")
+				annotationOk = false
+				break
+			} else {
+				nsInfo.routingExternalGWs = append(nsInfo.routingExternalGWs, parsedAnnotation)
+			}
+		}
+		if !annotationOk {
+			nsInfo.routingExternalGWs = nil
```

Comment:
(you don't actually need annotationOK; you could just do `nsInfo.routingExternalGWs = nil; break` rather than `annotationOk = false; break` above)